app/knight/knight.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+
+    def armour(self, armour: list) -> int:
+        for part in armour:
+            self.hp += part["protection"]
+        return self.hp
+
+    def weapon(self, weapon_power: int) -> int:
+        self.power += weapon_power
+        return self.power
+
+    def potion(self, potion: dict) -> callable:
+        if potion is not None:

```

Comment:
Due to your type annotations it always dict. Add None or Optional to potion type hinting